hail/python/hail/methods/qc.py

```diff
@@ -491,13 +504,128 @@ def n_discordant(counter):
     return glob, per_sample.cols(), per_variant.rows()
 
 
+def _service_vep(ht, config, block_size, csq, tolerate_parse_error):
+    token = secret_alnum_string(16)
+
+    backend = hl.context.current_backend()
+    assert isinstance(backend, hl.backend.service_backend.ServiceBackend)
+
+    tmp_dir = hl.backend.service_backend.tmp_dir()
+
+    vep_input_path = f'{tmp_dir}/vep/inputs/{token}.vcf'
+    vep_output_path = f'{tmp_dir}/vep/outputs/{token}'
+
+    hl.export_vcf(ht, vep_input_path, parallel='header_per_shard')
+
+    with hl.hadoop_open(config, 'r') as f:
+        local_config = json.loads(f.read())
+        image = local_config['image']
+        data_bucket = local_config['data_bucket']
+        data_mount = local_config['data_mount']
+        region = local_config['region']
+        env = local_config['env']
+
+    if csq:
+        vep_typ = hl.tstr
+    else:
+        vep_json_schema = local_config.get('vep_json_schema')
```

Comment:
@tpoterba What's the correct & forwards compatible way to serialize a Hail type as a string which we'll store somewhere and load in Python?

Reply:
I think we should just inline the type actually. this should do it:
```
hl.tstruct(
            assembly_name=hl.tstr,
            allele_string=hl.tstr,
            ancestral=hl.tstr,
            colocated_variants=hl.tarray(hl.tstruct(
                aa_allele=hl.tstr,
                aa_maf=hl.tfloat,
                afr_allele=hl.tstr,
                afr_maf=hl.tfloat,
                allele_string=hl.tstr,
                amr_allele=hl.tstr,
                amr_maf=hl.tfloat,
                clin_sig=hl.tarray(hl.tstr),
                end=hl.tint32,
                eas_allele=hl.tstr,
                eas_maf=hl.tfloat,
                ea_allele=hl.tstr,
                ea_maf=hl.tfloat,
                eur_allele=hl.tstr,
                eur_maf=hl.tfloat,
                exac_adj_allele=hl.tstr,
                exac_adj_maf=hl.tfloat,
                exac_allele=hl.tstr,
                exac_afr_allele=hl.tstr,
                exac_afr_maf=hl.tfloat,
                exac_amr_allele=hl.tstr,
                exac_amr_maf=hl.tfloat,
                exac_eas_allele=hl.tstr,
                exac_eas_maf=hl.tfloat,
                exac_fin_allele=hl.tstr,
                exac_fin_maf=hl.tfloat,
                exac_maf=hl.tfloat,
                exac_nfe_allele=hl.tstr,
                exac_nfe_maf=hl.tfloat,
                exac_oth_allele=hl.tstr,
                exac_oth_maf=hl.tfloat,
                exac_sas_allele=hl.tstr,
                exac_sas_maf=hl.tfloat,
                id=hl.tstr,
                minor_allele=hl.tstr,
                minor_allele_freq=hl.tfloat,
                phenotype_or_disease=hl.tint32,
                pubmed=hl.tarray(hl.tint32),
                sas_allele=hl.tstr,
                sas_maf=hl.tfloat,
                somatic=hl.tint32,
                start=hl.tint32,
                strand=hl.tint32)),
            context=hl.tstr,
            end=hl.tint32,
            id=hl.tstr,
            input=hl.tstr,
            intergenic_consequences=hl.tarray(hl.tstruct(allele_num=hl.tint32,
                                                         consequence_terms=hl.tarray(hl.tstr),
                                                         impact=hl.tstr,
                                                         minimised=hl.tint32,
                                                         variant_allele=hl.tstr)),
            most_severe_consequence=hl.tstr,
            motif_feature_consequences=hl.tarray(hl.tstruct(allele_num=hl.tint32,
                                                            consequence_terms=hl.tarray(hl.tstr),
                                                            high_inf_pos=hl.tstr,
                                                            impact=hl.tstr,
                                                            minimised=hl.tint32,
                                                            motif_feature_id=hl.tstr,
                                                            motif_name=hl.tstr,
                                                            motif_pos=hl.tint32,
                                                            motif_score_change=hl.tfloat,
                                                            strand=hl.tint32,
                                                            variant_allele=hl.tstr)),
            regulatory_feature_consequences=hl.tarray(hl.tstruct(allele_num=hl.tint32,
                                                                 biotype=hl.tstr,
                                                                 consequence_terms=hl.tarray(hl.tstr),
                                                                 impact=hl.tstr,
                                                                 minimised=hl.tint32,
                                                                 regulatory_feature_id=hl.tstr,
                                                                 variant_allele=hl.tstr)),
            seq_region_name=hl.tstr,
            start=hl.tint32,
            strand=hl.tint32,
            transcript_consequences=hl.tarray(hl.tstruct(allele_num=hl.tint32,
                                                         amino_acids=hl.tstr,
                                                         biotype=hl.tstr,
                                                         canonical=hl.tint32,
                                                         ccds=hl.tstr,
                                                         cdna_start=hl.tint32,
                                                         cdna_end=hl.tint32,
                                                         cds_end=hl.tint32,
                                                         cds_start=hl.tint32,
                                                         codons=hl.tstr,
                                                         consequence_terms=hl.tarray(hl.tstr),
                                                         distance=hl.tint32,
                                                         domains=hl.tarra(hl.tstruct(db=hl.tstr,
                                                                                     name=hl.tstr)),
                                                         exon=hl.tstr,
                                                         gene_id=hl.tstr,
                                                         gene_pheno=hl.tint32,
                                                         gene_symbol=hl.tstr,
                                                         gene_symbol_source=hl.tstr,
                                                         hgnc_id=hl.tstr,
                                                         hgvsc=hl.tstr,
                                                         hgvsp=hl.tstr,
                                                         hgvs_offset=hl.tint32,
                                                         impact=hl.tstr,
                                                         intron=hl.tstr,
                                                         lof=hl.tstr,
                                                         lof_flags=hl.tstr,
                                                         lof_filter=hl.tstr,
                                                         lof_info=hl.tstr,
                                                         minimised=hl.tint32,
                                                         polyphen_prediction=hl.tstr,
                                                         polyphen_score=hl.tfloat,
                                                         protein_end=hl.tint32,
                                                         protein_start=hl.tint32,
                                                         protein_id=hl.tstr,
                                                         sift_prediction=hl.tstr,
                                                         sift_score=hl.tfloat,
                                                         strand=hl.tint32,
                                                         swissprot=hl.tstr,
                                                         transcript_id=hl.tstr,
                                                         trembl=hl.tstr,
                                                         uniparc=hl.tstr,
                                                         variant_allele=hl.tstr)),
            variant_class=hl.tstr)
```